src/hamiltonian/Constraint.js

```diff
@@ -31,6 +31,50 @@ class Constraint {
 	get parameters(){
 		return this.conf
 	}
+
+	/** Get a cellid or cellkind-specific parameter for a constraint. 
+	 * This usually refers to @function paramOfKind, which returns
+	 * this.conf[param][cellkind], and thus assumes that the parameter is 
+	 * indexable by cellkind.
+	 * If CPMEvol is used this is redirected to @function paramOfCell
+	 * which looks whether the parameter is overwritten in an @object Cell
+	 * and otherwise returns @function paramOfKind
+	 * 
+	 * @param {string} param - name of parameter in conf object
+	 * @param {CellId} cid - Cell Id of cell in question, if id-specific parameter is not present, cellkind of cid is used
+	@return {any} parameter - the requested parameter
+	*/
+	cellParameter(param, cid){
+		return this.paramOfKind(param, cid)
+	}
```

Comment:
I think this works nicely now, but note that this function is overwritten by the constructor in either case (either by paramOfCell or by paramOfKind). I would suggest some small changes to the doc-block to make this more clear
```suggestion
	/** Get a cellid or cellkind-specific parameter for a constraint. 
	 * This function is here to document its functionality, but it is 
	 * always overwritten by the constructor (via "set CPM") to point
	 * to another function. This is normally  @function  paramOfKind,
	 * which retrieves the parameter from the conf object for the current {@link cellKind}.
	 * If CPMEvol is used this is instead redirected to @function  paramOfCell,
	 * which looks whether the parameter is overwritten in an @object Cell
	 * and otherwise returns @function paramOfKind
	 * @abstract
	 * 
	 * @param {string} param - name of parameter in conf object
	 * @param {CellId} cid - Cell Id of cell in question, if id-specific parameter is not present, cellkind of cid is used
	@return {any} parameter - the requested parameter
	*/
	/* eslint-disable no-unused-vars */
	cellParameter(param, cid){
		throw( "this is a template function that should never actually be called as it is overwritten to point to paramOfCell() or paramOfKind().")
	}
```

Reply:
I agree that this is the intended use, and committed this rewrite. However, multiple jasmine tests fail on this throw, which I'll investigate further. The setting of CPM is not technically a constructor, but is called through bind(), right? Is it possible that some tests already call for parameters before setting the CPM?

Reply:
Found the issue: some constraints overwrite set CPM internally, and do not call a super implementation

Reply:
Fixed all tests with this implementation. There were two underlying issues: some CPM setters were overwritten in subclasses, and the LocalConnectivitySpec test directly set the .C private variable of the constraint instead of going through the CPM settter.

Reply:
> > Thanks - this is some really nice work!
> > Before merging, I have a few small questions/comments. In particular I wanted to ask you about the colormap functionality:
> > 
> > * Is this something you wrote yourself or did you get it somewhere?
> > * Is there a reason why you have included files in the "build" folder (`colormap.js` and `colormap-cjs.js`) _and_ incorporated them into the Artistoo build?
> > 
> > As colormaps are an entirely different functionality that do not have much to do with CPM models, I would suggest removing them from the main build and using a similar construction as with the `fpsmeter.min.js`: we can add the separate file(s) (e.g. in the examples folder) and then load in the HTML/Node scripts where appropriate.
> > I also had some (minor) more specific questions and comments which I included with the relevant code. Could you have a look at these before I merge the pull request? Thanks!
> 
> On the colormaps, this is indeed partially rewritten code from somewhere else, I planned to attribute this in the comments, but forgot. I'll do this before merge.
> And I can remove the colormap from the full build as it is not required, however, I would suggest also moving the fpsmeter to the accessories folder and including it in the full build, and making this a folder for small add-ons that are frequently necessary for visualization and extendability. In this way the example folder also only gets a single .js file, and sharing a simulation that runs in the browser does not require multiple accessory files. What do you think of this?

I get your point that it would be convenient not to have to load too many files, but I still think it is neater to keep these things separate from the main library. Having fpsmeters or colormaps is not technically necessary for sharing a simulation, and people may also wish to use their own solutions if they want to do these things -- so perhaps it is better not to include these in the main `artistoo.js` and `artistoo-cjs.js` files. That said, perhaps we can bundle some of these additional things in a separate file `viztools` or something. That way, you'd have to load an extra file but it would be at most one. 

I'll raise an issue for this and think a bit more on what would be the neatest solution. Until then, perhaps we can keep the colormap.js the same way we have the fpsmeter now?